src/Common/CgroupsMemoryUsageObserver.cpp

```diff
@@ -214,21 +214,30 @@ uint64_t CgroupsMemoryUsageObserver::File::readMemoryUsage() const
             ///   rss 15
             ///   [...]
             std::string key;
+            bool found_rss = false;
+
             while (!buf.eof())
             {
                 readStringUntilWhitespace(key, buf);
                 if (key != "rss")
                 {
                     std::string dummy;
                     readStringUntilNewlineInto(dummy, buf);
+                    buf.ignore();
                     continue;
                 }
+
                 assertChar(' ', buf);
                 readIntText(mem_usage, buf);
                 assertChar('\n', buf);
+                found_rss = true;
```

Comment:
After some local testing, I deleted l. 232  for the (obscure but possible) case that the file contains a single line with key `rss` but no newline. Now, l. 232 is back, I guess unintentional because of force-pushing.

@kitaisreal Like to remove l. l. 232 again? Thanks.

Reply:
Yes. Updated.